### PR TITLE
Refactor ConsumerRecordHandler

### PR DESCRIPTION
**Pull request by slinkydeveloper**
This PR is meant as a first step for #589

## Proposed Changes

- :broom: Reimplemented ConsumerRecordHandler (now named RecordDispatcher) to return a Future when dispatch is completed
- :broom: Some name changes and some classes moved
- :gift: Now OffsetManager methods are async

### CODE REVIEW

File: data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/RecordDispatcher.java [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+      |    | filter not |   +---------->+  failure     |     |  success    +---->end
+      |    | matching   |               +----+---------+     |             |
+      |    |            |                    |               +-----+-------+
+      |    +---+--------+              +-----v-------+             |
+      |        |                       | dlq failure |             v
+      |        |                       +-------------+----------> end
+      +->end<--+
+
+
+     */
```
Comment (pierDipi) [inline data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/RecordDispatcher.java:139]: Awesome :heart_eyes: 

Nit, some extra empty lines:
```suggestion
    /*
    That's pretty much what happens here:

    +------------+    +------------+     +-------------+
    |            |    |            |     |             |
    |  record    |    |  filter    |     |  subscriber |
    |  received  +--->+  matching  +---->+  success    +-->end
    |            |    |            |     |             |
    +-+-------+--+    +-----+------+     +-----+-------+
      |       |             |                  |
      |       v             |                  |             +-------------+
      |    +--+---------+   |           +------v-------+     |             |
      |    |            |   |           |  subscriber  +---->+  dlq        |
      |    | filter not |   +---------->+  failure     |     |  success    +---->end
      |    | matching   |               +----+---------+     |             |
      |    |            |                    |               +-----+-------+
      |    +---+--------+              +-----v-------+             |
      |        |                       | dlq failure |             v
      |        |                       +-------------+----------> end
      +->end<--+
     */
```

**Review by pierDipi — APPROVED** [review]
This is great! A small comment, feel free to ignore.

/lgtm

**Comment by slinkydeveloper** [comment]
@pierDipi I've rebased with your fix for the test flakyness :)

**Comment by pierDipi** [comment]
I don't see https://github.com/knative-sandbox/eventing-kafka-broker/pull/557/files#diff-f381c3ca71d3199724fad85a234c55b0b0cfecf7db83db13e1bd06c6e970f9bdR68

/lgtm

**Comment by slinkydeveloper** [comment]
I'm sorry, you're right...

**Review by pierDipi — APPROVED** [review]
/lgtm

**Pull request merged**